README.md

```diff
@@ -1,2 +1,233 @@
+| /foo/bar   | /foo/baz  |    ---     | /foo/baz     | The file's name will be changed from bar to baz.                       |
+| /foo1/bar  | /foo2/    |    True    | /foo/bar     | The file will be copied to '/foo2/'                                    |
+| /foo/bar/  | /foo/     |    True    | /foo/        | The content of '/foo/bar/' will be copied to '/foo/'                   |
+| /foo1/bar/ | /foo2/    |   False    | /foo2/bar/   | '/foo1/bar/' will be copied recursively to '/foo2/bar/'                |
+| /foo1/bar/ | /foo2/baz |    ---     |     ---      | ValueError will be raised in attempting to copy directory in to a file |
+
+```python
+>>> pyminio_client.cp('/foo/bar', '/foo/baz')
+>>> pyminio_client.cp('/foo1/bar', '/foo2/')
+>>> pyminio_client.cp('/foo/bar/', '/foo/', recursive=True)
+>>> pyminio_client.cp('/foo1/bar/', '/foo2/', recursive=True)
+>>> pyminio_client.cp('/foo1/bar/', '/foo2/baz', recursive=True)
+ValueError: can not activate this method from directory to a file.
+```
+
+### mv
+Pyminio.mv works like [cp](#cp) only that it remove the source after complete the transfare. Works like linux's 'mv (-r)'.
+
+This func can only move recursively when the recursive flag is True. If not, it will raise a ValueError.
+
+```python
+>>> pyminio_client.mv('/foo/bar/', '/foo/baz/')
+```
+
+### get
+Pyminio.get return an object from given path. This object will return as a pyminio.File object or an pyminio.Folder object, that both inherit from pyminio.ObjectData
+
+This objects will contain metadata, their path and name.
+If its a File object it will contains it's data.
+
+```python
+>>> pyminio_client.get('/foo/bar/baz')
+File(name='baz', full_path='/foo/bar/baz', metadata=AttrDict({'is_dir': False, 'last_modified': time.struct_time(...), 'size': ..., 'content-type': ...}), data=...)
+```
+
+### get_last_object
+Pyminio.get_last_object will return the last modified object inside a given directory.
+
+This function must get a directory path or it will raise a ValueError.
+
+```python
+>>> pyminio_client.get_last_object('/foo/bar/')
+File(name='baz', full_path='/foo/bar/baz', metadata=AttrDict({'is_dir': False, 'last_modified': time.struct_time(...), 'size': ..., 'content-type': ...}), data=...)
+```
+
+### put_data
+Pyminio.put_data gets a path, data in bytes, and some metadata, and create an object inside the given path.
+
+```python
+>>> data = b'test'
+>>> metadata = {'Pyminio-is': 'Awesome'}
+>>> pyminio_client.put_data(path='/foo/bar/baz', data=data, metadata=metadata)
+```
+
+### put_file
+Pyminio.put_file works like [put_data](#put_data) only that instead of data it gets a path to a file in you computer. Then it will copy this file to the given location.
+
+```python
+>>> metadata = {'Pyminio-is': 'Awesome'}
+>>> pyminio_client.put_data(path='/foo/bar/baz', file_path='/mnt/some_file', metadata=metadata)
```

Comment:
`path` and `file_path` is very confusing
you need to make sure the user understands which is the source and which is the target
make sure the source is the first one (just like Linux)
also, you may be better using keyword-only arguments, like so:
```python
def put_data(*, source_path, dest_path, metadata)
```

Reply:
d